File: javascript-node/samples/getSecurityPriceTechnicalsRsi.js
var intrinioSDK = require('intrinio-sdk');
intrinioSDK.ApiClient.instance.authentications['ApiKeyAuth'].apiKey = "YOUR API KEY";
 
var securityAPI = new intrinioSDK.SecurityApi();
 
var identifier = "AAPL"; // String | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)
 
var opts = {};

securityAPI.getSecurityPriceTechnicalsRsi(identifier, opts).then(function(data) {
  var indicator = data.indicator;
  var security = data.security;
  var technicals = data.technicals;

  console.log('Technicals for ' + security.ticker);
  console.log(technicals.length + ' values for ' + indicator.name + ' returned!');
  console.log();

  technicals.forEach(function(technical){
    console.log("DateTime: " + technical.date_time);
    console.log("RSI:      " + technical.value);
    console.log("--------------------------------------------------------------------------------");
  });
  
}, function(error) {
  console.error(error);
});

File: java/JavaSDKExample/src/main/java/getSecurityPriceTechnicalsRsi.java
package com.intrinio.javasdkexample;

import com.intrinio.invoker.ApiClient;
import com.intrinio.invoker.ApiException;
import com.intrinio.invoker.Configuration;
import com.intrinio.invoker.auth.*;
import com.intrinio.api.SecurityApi;
import com.intrinio.models.*;

import java.util.List;

public class App
{
  public static void main( String[] args )
  {
    ApiClient defaultClient = Configuration.getDefaultApiClient();
    ApiKeyAuth auth = (ApiKeyAuth) defaultClient.getAuthentication("ApiKeyAuth");
    auth.setApiKey("YOUR API KEY");

    SecurityApi securityApi = new SecurityApi();

    String identifier = "AAPL"; // String | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)
    Integer period = null; // Integer | The number of observations, per period, to calculate Relative Strength Index
    String priceKey = null; // String | The Stock Price field to use when calculating Relative Strength Index
    String startDate = null; // String | Return technical indicator values on or after the date
    String startTime = null; // String | Return prices at or after this time (24-hour)
    String endDate = null; // String | Return technical indicator values on or before the date
    String endTime = null; // String | Return prices at or before this time (24-hour)
    String timezone = null; // String | Returns technical indicators in this timezone
    Integer pageSize = null; // Integer | The number of results to return
    String nextPage = null; // String | gets the next page of data from an already-executed API call

    try {
      ApiResponseSecurityTechnicalsSingleValue result = securityApi.getSecurityPriceTechnicalsRsi(identifier, period, priceKey, startDate, startTime, endDate, endTime, timezone, pageSize, nextPage);
      SecuritySummary security = result.getSecurity();
      Indicator indicator = result.getIndicator();
      List<Technical> technicals = result.getTechnicals();

      System.out.println("Technicals for " + security.getTicker());
      System.out.println(technicals.size() + " values for " + indicator.getName() + " returned!");
      System.out.println();

      for (Technical technical : technicals) {
        System.out.println("DateTime: " + technical.getDateTime());
        System.out.println("RSI:      " + technical.getValue());
        System.out.println("-----------------------------------------------------------------------");
      }
    } catch (ApiException e) {
      System.err.println("Exception when calling SecurityApi#getSecurityPriceTechnicalsRsi");
      e.printStackTrace();
    }
  }
}
